### DISCUSSION TITLE
Tasmota 12.4 - Rule won't trigger on Button3#state vs. Switch3#state

### DISCUSSION THREAD

**Discussion by Saxin** [arendst/Tasmota#18404] [Q&A]
Hi guys, hope someone can help.

Device is LoraTap [https://templates.blakadder.com/loratap_SC500W.html](url)
The FW is compiled using TasmoCompiler (gitpod) based on 12.4:
"Program Version | 12.4.0(TasmoCompiler-esp8266generic)
2023-04-12T10:03:56
2_7_4_9/2.2.2-dev(38a443e)"

**GPIO13 is configured as Button 3, Options 1,11,13 are at default FW values. ButtonTopic is 0.**
(GPIO is pulled HIGH pullup, when button pressed -> pulled to GND.)

Single Rule issued:
**Backlog Rule3 ON Button3#state DO Publish cmnd/XXXXX/POWER %value% endon; Rule3 1**
XXXXX - is topic of device to be controlled, it's single gang light switch also running Tasmota.

I've confirmed the Publish command is correct by issuing it directly into console.

This isn't working for me, no mqtt command is issued, I'm not sure how to confirm the press is triggered ?

Now changing the button to switch 3, adding dummy relay 3 in template and issuing:
**backlog SwitchMode3 4; Rule3 on Switch3#state do Publish cmnd/tasmota_XXXXX/POWER %value% endon; Rule3 1**

Everything works as expected.
Breaking my head here, why the button version doesn't seem to work.

Thanks guys.

**Comment by barbudor** [comment]
```
20:06:43.692 CMD: Backlog Rule3 ON Button3#state DO Publish cmnd/XXXXX/POWER %value% endon; Rule3 1
20:06:43.712 RUL: Stored uncompressed, would compress from 58 to 46 (-21%)
20:06:43.717 MQT: stat/nodemcu/RULE = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":58,"Free":453,"Rules":"ON Button3#state DO Publish cmnd/XXXXX/POWER %value% endon"}}
20:06:43.945 MQT: stat/nodemcu/RULE = {"Rule3":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":58,"Free":453,"Rules":"ON Button3#state DO Publish cmnd/XXXXX/POWER %value% endon"}}
20:07:07.321 RUL: BUTTON3#STATE performs "Publish cmnd/XXXXX/POWER 2"
20:07:07.331 MQT: cmnd/XXXXX/POWER = 2
20:07:09.428 RUL: BUTTON3#STATE performs "Publish cmnd/XXXXX/POWER 2"
20:07:09.437 MQT: cmnd/XXXXX/POWER = 2
20:07:10.181 RUL: BUTTON3#STATE performs "Publish cmnd/XXXXX/POWER 2"
20:07:10.191 MQT: cmnd/XXXXX/POWER = 2
```

¯\_(ツ)_/¯

Tested on  12.4.0 and 12.4.0.5
Officially, all index should start at 1 so you shouldn't use Button_3 without a Button_1 and a Button_2
However there have been some recent change that can allow that in some cases (but still not recommended)

**Reply by sfromis** [reply]
And better check your log for details, after activating `Weblog 3`
